### PR TITLE
Fix autoConnect for MockConnector

### PR DESCRIPTION
**Pull request by Volchunovich**
## Description

[Issue](https://github.com/wagmi-dev/wagmi/issues/1688)

## Additional Information

- [x] I read the [contributing docs](/wagmi-dev/references/blob/main/.github/CONTRIBUTING.md) (if this is your first contribution)

Your ENS/address: 0xa37a691C3D25cc91aEd2A3CDb3FCe9878657bA05

### CODE REVIEW

**Comment by Volchunovich** [comment]
@tmm @jxom could you review this PR, please?

File: .changeset/fast-crabs-pay.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@wagmi/connectors": minor
```
Comment (tmm) [inline .changeset/fast-crabs-pay.md:2]: Backwards compatible bug fix

```suggestion
"@wagmi/connectors": patch
```

Comment (Volchunovich) [reply .changeset/fast-crabs-pay.md]: fixed

**Comment by tmm** [comment]
Pulled this into the main repo and there are a few tests in [`packages/core/src/config.test.ts`](https://github.com/wagmi-dev/wagmi/blob/main/packages/core/src/config.test.ts) that fail. Mind taking a look at those?

**Comment by Volchunovich** [comment]
> Pulled this into the main repo and there are a few tests in [`packages/core/src/config.test.ts`](https://github.com/wagmi-dev/wagmi/blob/main/packages/core/src/config.test.ts) that fail. Mind taking a look at those?

Hmm, I'll check it.

**Comment by Volchunovich** [comment]
@tmm I checked config.test.ts, and fixed isAuthorized and [autoConnect tests in @wagmi/core](https://github.com/wagmi-dev/wagmi/pull/2485/files#diff-f17f1fbd33c1c38cfaa708c4a1e95a2305973d3e1720fcd50985fc1ed89c4597R114). What do you think is it okay? This happen, because now you're not connected as always and if you want to autoConnect you should be connected before.

**Comment by tmm** [comment]
This PR can be re-opened on the [Wagmi repo](https://github.com/wagmi-dev/wagmi). Connectors are moving over there.

**Pull request closed without merging**